test/lib/code_corps/donation_goals_manager_test.exs

```diff
@@ -0,0 +1,171 @@
+defmodule CodeCorps.DonationGoalsManagerTest do
+  use ExUnit.Case, async: true
+
+  use CodeCorps.ModelCase
+
+  alias CodeCorps.DonationGoal
+  alias CodeCorps.DonationGoalsManager
+
+  defp assert_current_goal_id(goal_id) do
+    current_goal = DonationGoal |> Repo.get_by(current: true)
+    assert current_goal.id == goal_id
+  end
+
+  defp donate(plan, amount) do
+    insert(:stripe_connect_subscription, quantity: amount, stripe_connect_plan: plan)
+  end
+
+  describe "create/1" do
+    test "inserts new goal, returns {:ok, record}" do
+      project = insert(:project)
+      insert(:stripe_connect_plan, project: project)
+
+      {:ok, %DonationGoal{} = donation_goal} = DonationGoalsManager.create(%{amount: 10, description: "Test", project_id: project.id})
+      assert donation_goal.current
+    end
+
+    test "returns {:error, changeset} if there are validation errors" do
+      {:error, %Ecto.Changeset{} = changeset} = DonationGoalsManager.create(%{amount: 10})
+      refute changeset.valid?
+    end
+
+    test "sets current goal correctly" do
+      project = insert(:project)
+      plan = insert(:stripe_connect_plan, project: project)
+
+      {:ok, first_goal} = DonationGoalsManager.create(%{amount: 10, description: "Test", project_id: project.id})
+
+      # total donated is 0,
+      # only goal inserted is the first goal
+      assert_current_goal_id(first_goal.id)
+
+      {:ok, second_goal} = DonationGoalsManager.create(%{amount: 20, description: "Test", project_id: project.id})
+
+      # total donated is still 0
+      # first goal larger than 0 is the very first goal
+      assert_current_goal_id(first_goal.id)
```

Comment:
I think maybe reversing amounts on these would be good, so `20` then `10`, so we know `10` becomes the goal.